packages/bot-strategy-runner/src/ConfigBuilder.ts

```diff
@@ -0,0 +1,203 @@
+      botIdentifier: `${syntheticSymbols[contractAddress]} ${botType}`,
+      botNetwork: config.botNetwork,
+      financialContractAddress: contractAddress,
+      errorRetries: addressConfig.errorRetries,
+      errorRetriesTimeout: addressConfig.errorRetriesTimeout,
+      pollingDelay: 0,
+      startingBlock: addressConfig.startingBlock ? addressConfig.startingBlock : process.env.STARTING_BLOCK_NUMBER,
+      endingBlock: addressConfig.endingBlock ? addressConfig.endingBlock : process.env.ENDING_BLOCK_NUMBER
+    };
+    if (botType == "liquidator") {
+      const botConfig: liquidatorConfig = {
+        ...commonConfig,
+        priceFeedConfig: addressConfig.priceFeedConfig,
+        liquidatorConfig: addressConfig.liquidatorConfig,
+        liquidatorOverridePrice: addressConfig.liquidatorOverridePrice
+      };
+      botConfigs.push(botConfig);
+    }
+
+    if (botType == "disputer") {
+      const botConfig: disputerConfig = {
+        ...commonConfig,
+        priceFeedConfig: addressConfig.priceFeedConfig,
+        disputerConfigConfig: addressConfig.disputerConfigConfig,
+        disputerOverridePrice: addressConfig.disputerOverridePrice
+      };
+      botConfigs.push(botConfig);
+    }
+    if (botType == "monitor") {
+      const botConfig: monitorConfig = {
+        ...commonConfig,
+        optimisticOracleAddress: addressConfig.optimisticOracleAddress,
+        monitorConfigConfig: addressConfig.monitorConfigConfig,
+        tokenPriceFeedConfig: addressConfig.tokenPriceFeedConfig,
+        medianizerPriceFeedConfig: addressConfig.medianizerPriceFeedConfig,
+        denominatorPriceFeedConfig: addressConfig.denominatorPriceFeedConfig
+      };
+      botConfigs.push(botConfig);
+    }
+  });
+  return botConfigs;
+}
+
+// Takes in two objects of any depth and merges them. In the case of a collision in keys the second object will take priority.
+export function mergeConfig(...args: any) {
+  const target: any = {}; // create a new object
+
+  // deep merge the object into the target object
+  const merger = (obj: any) => {
+    for (const prop in obj) {
+      if ({}.hasOwnProperty.call(obj, prop)) {
+        if (Object.prototype.toString.call(obj[prop]) === "[object Object]") {
+          // if the property is a nested object
+          target[prop] = mergeConfig(target[prop], obj[prop]);
+        } else {
+          // for regular property
+          target[prop] = obj[prop];
+        }
+      }
+    }
+  };
+
+  // iterate through all objects and deep merge them with target
+  for (let i = 0; i < args.length; i++) {
+    merger(args[i]);
+  }
+
+  return target;
+}
+
+// Takes in an object of any structure and returns the exact same object with all strings converted to check sum format.
+function replaceAddressCase(object: any) {
+  const stringifiedObject = JSON.stringify(object);
+  const replacedStringifiedObject = stringifiedObject.replace(/0x[a-fA-F0-9]{40}/g, toChecksumAddress);
```

Comment:
I don't think this will cause any problems, but is it possible that this will mess with byte strings? I'm assuming that even if it does, it won't break anything. Just wanted to mention it.

Reply:
ye, it _could_ but I think that this will only modify strings of length 40 starting with 0x. bytestrings tend to be longer at 64 chars (bytes32) with a 0x at the front.

Reply:
I think it will touch any strings of length _greater than_ 40 hex characters (since the first 40 hex characters would satisfy the regex). Regardless, I'm not too worried about checksumming random byte strings > 40 characters -- just might look a little weird in the logs.

My only point was to raise the issue in case you thought this _could_ break something (like toChecksum erroring on non-addresses or common interpreters breaking if we paste in bytes with varied casings).